exercises/practice/armstrong-numbers/.meta/lib/example.dart

```diff
@@ -3,17 +3,16 @@ import 'dart:math' show pow;
 class ArmstrongNumbers {
   /// The parameters and variables within the method that are set to final in order to prevent the accidentally modification of the value.
   /// As those variables are not needed to be changed.
-  bool isArmstrongNumber(final num input) {
-    final String numberAsString = input.toString();
-    final int numOfDigits = numberAsString.length;
-    num sum = 0;
+  bool isArmstrongNumber(final String input) {
+    final numOfDigits = input.length;
+    var sum = BigInt.from(0);
 
-    for (int count = 0; count < numOfDigits; count++) {
-      final String digitAsString = numberAsString.substring(count, count + 1);
-      final int digit = int.parse(digitAsString);
-      sum += pow(digit, numOfDigits);
+    for (var count = 0; count < numOfDigits; count++) {
+      final digitAsString = input.substring(count, count + 1);
+      final digit = BigInt.parse(digitAsString);
+      sum += digit.pow(numOfDigits);
     }
 
-    return input == sum;
+    return input == sum.toString();
```

Comment:
Since you mentioned _approaches_ in #433 

What do you think of an approach which uses `int` for small enough numbers & switches `BigInt` for big ones?
- This could be accomplished by checking the input length. My rational behind this approach is that `BigInt`s are generally very very slow compared to `int`. 
- So if you were to build a program that deals with Money, or numbers in general & your numbers are generally small enough to fit in 64 bits. 
- This approach would make your program fast for most of the common use cases and daily needs. 
- And once a while when your program needs to deal with very large numbers, it switches to `BigInt`s, the performance will be noticeably slow for certain calculations but the end-users would expect that, since they have some understanding that big numbers are being crunched.

While not necessarily related to `armstrong-numbers`. I think this could benefit the learners.
I am curious to know whats your take on this?

Reply:
Yepp. I think it's a great example of an interesting approach that would be worth exploring in a dig deeper article. Ruby does something similar with strings. The underlying C implementation uses one implementation for shorter strings and another for longer strings, if I recall correctly, for similar reasons.

Here's an example of approaches explored in C# for the "bob" exercise, including an additional article about performance trade-offs between different solutions: https://exercism.org/tracks/csharp/exercises/bob/dig_deeper (you might have to look at it in a private browser tab; not sure).